### PR TITLE
fix compile warning on gleam 1.2+, gleam 1.1 formatter, min gleam 1.0, deps update

### PR DESCRIPTION
**Pull request by inoas**

### CODE REVIEW

File: src/gleam_community/colour/accessibility.gleam [hunk 1]
```diff
@@ -55,9 +55,9 @@ import gleam_community/colour.{type Colour}
 
 fn intensity(colour_value: Float) -> Float {
   // Calculation taken from https://www.w3.org/TR/WCAG20/#relativeluminancedef
-  case True {
```
Comment (inoas) [inline src/gleam_community/colour/accessibility.gleam:58]: this PR fixes this compiler warning

File: src/gleam_community/colour.gleam [hunk 2]
```diff
@@ -701,8 +701,10 @@ pub fn to_hsla(colour: Colour) -> #(Float, Float, Float, Float) {
 pub fn to_css_rgba_string(colour: Colour) -> String {
   let #(r, g, b, a) = to_rgba(colour)
 
-  let percent = fn(x: Float) -> Float {
+  let percent = fn(
+    x: Float,
     // This won't fail because we are always dividing by 100.0
```
Comment (aslilac) [inline src/gleam_community/colour.gleam]: huh, weird formatting. and it mangled this comment.

File: src/gleam_community/colour.gleam [hunk 3]
```diff
@@ -713,8 +715,10 @@ pub fn to_css_rgba_string(colour: Colour) -> String {
     p
   }
 
-  let round_to = fn(x: Float) -> Float {
+  let round_to = fn(
+    x: Float,
     // This won't fail because we are always dividing by 1000.0
```
Comment (aslilac) [inline src/gleam_community/colour.gleam]: same thing here

**Comment by hayleigh-dot-dev** [comment]
We won't be accepting changes focused on gleam nightly, because they will fail our CI and could cause confusion for folks using the latest stable gleam release.

**Comment by inoas** [comment]
> We won't be accepting changes focused on gleam nightly, because they will fail our CI and could cause confusion for folks using the latest stable gleam release.

I am not sure why the CI was read before, but I am now using gleam 1.1 for any public library contribution.

The warning appears on nightly, but the change is tested locally on gleam 1.1

**Comment by inoas** [comment]
The warning appears on nightly, but the change is tested locally on gleam 1.1

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+- Updated dependencies and code to be in line with Gleam 1.0.
+- Reformat with Gleam 1.1.
+- Enable testing against the Javascript target in the CI.
```
Comment (aslilac) [inline CHANGELOG.md]: I think the changelog should mostly be for user facing things. "Updated dependencies" is really the only bit out of all of this with any externally observable impact.

Comment (inoas) [reply CHANGELOG.md]: I hope it is more clear now.

File: src/gleam_community/colour.gleam [hunk 5]
```diff
@@ -239,12 +239,12 @@ fn rgba_to_hsla(
 
   let max_colour = float.max(r, float.max(g, b))
 
-  let h1 = case True {
-    _ if max_colour == r -> float.divide(g -. b, max_colour -. min_colour)
-    _ if max_colour == g ->
+  let h1 = case max_colour == r, max_colour == g {
```
Comment (inoas) [inline src/gleam_community/colour.gleam:242]: I kind of wish we had elixir's cond but with if multiple clause and else syntax.
the old case way mimicked that.

**Comment by aslilac** [comment]
I'm fine with running tests in CI against multiple runtimes, but there's a lot of other random changes lumped in that feel like stylistic concerns, and that diverge from the standard template that all of the gleam-community projects use

**Comment by inoas-nbw** [comment]
> I'm fine with running tests in CI against multiple runtimes, but there's a lot of other random changes lumped in that feel like stylistic concerns, and that diverge from the standard template that all of the gleam-community projects use

for the ci or for anything else

> - run: gleam publish --yes
this?

it is mostly from a new gleam app.

**Comment by inoas** [comment]
I have brought up the topic of abusing case in elixir cond style and it is possible again in 1.2 rc2 so I will close this for now.

**Pull request closed without merging**